app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -31,12 +33,45 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
             FilterTransformers.rawFilters = angular.copy(filters);
             // Remove set filter as it is only relevant to collections and should be immutable in that view
             delete activeFilters.set;
-            $scope.activeFilters = _.mapObject(activeFilters, makeArray);
+            var savedSearchEntity = PostFilters.getModeEntity();
+            if ($scope.savedSearch && savedSearchEntity.id !== $scope.savedSearch.id) {
+                $scope.savedSearch = savedSearchEntity;
+            } else if (!$scope.savedSearch && PostFilters.getMode() === 'savedsearch') {
+                $scope.savedSearch = savedSearchEntity;
+            }
+
+            /**
+             * This handles the requirement to have saved search filters displayed in a different way
+             * from the rest of the filters.
+             *  - If there is no saved_search in the filters, continue with the normal filters (else)
+             *  - If there is a saved search:
+             *  - - set it in the scope : this is what we will use to display the saved search and its filters in the UI
+             *  - - get clean version (no defaults) of the saved search filters
+             *  - - get a clean activeFilters array that does not include the saved search filters. $scope.activeFilters: this
+             *  is the array we will be using to show the "extra" filters the user can set AFTER they selected  a saved search
+             *  (keep in mind that when a saved search is selected all filters are erased in favor of the saved search)
+             *  - - - Diffing rules: value from activeFilters takes priority over value from search. this is because
+             *  the user will always have selected the value after they select a saved search,meaning they want to change it.
+             *  - - - Diffing rules: if the value is equal, just ignore it/send empty value.
+             *  - - - Diffing rules: if the value is a number or string, return Value because it is a priority (the one from filters not from saved search)
+             *  - - - Diffing rules: return the _difference if the value is not equal, not a number or not a string, because we will want to show for isntance:
+             *  saved search: tag id 1 + filters tag id 2 (so it's not just ignoring the arrays)
+             */
+            activeFilters = _.mapObject(activeFilters, makeArray);
+            if ($scope.savedSearch) {
+                // get clean version (no defaults) of the saved search filters
+                $scope.savedSearch.filter = _.mapObject(PostFilters.getCleanActiveFilters($scope.savedSearch.filter), makeArray);
+                $scope.activeFilters = _.mapObject(activeFilters, function (value, key) {
+                    return _.difference(value, $scope.savedSearch.filter[key]);
+                });
+            } else {
+                $scope.activeFilters = activeFilters;
```

Comment:
I set it here instead of pre-setting it just because we watch active filters, no need to change it in line 65 and then again in the saved search if . But might be one of those scenarios where we prefer to do it on line 65 because I admit it'd be more readable/easier to follow. Not sure